angularSTEP/src/app/recipe-card/recipe-card.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+interface recipeData{
+  recipeName: string,
+  description: string,
+  difficulty: string
+}
+
+@Component({
+  selector: 'app-recipe-card',
+  templateUrl: './recipe-card.component.html',
+  styleUrls: ['./recipe-card.component.scss']
+})
+export class RecipeCardComponent implements OnInit{
+
+  @Input() recipe!: recipeData;
+  recipeName!: string;
```

Comment:
I'm still leaning towards avoiding these extra class fields+assignments just to remove the "recipe." part from the templating, but this is fine if there's a good reason to do all the extra work here. I mostly want to avoid doing things the hard way for the only reason of "we do things the hard way"

Reply:
Yes ^ I actually went back and changed my code too. My bad for making it a little harder than we needed it too. You can just add `*ngIf='recipe'` to the container div so it will only show if the recipe is not null, which will help in making sure there are no errors.

Reply:
Im still a bit confused on what adding *ngIf would chang.

Reply:
`*ngIf='recipe'` will check if the recipe variable is `null`. This will make sure that within the div, when you are accessing recipe's fields, you aren't trying to access fields from a null variable.